### ISSUE TITLE
[BUG]: Incorrect return type hint of Phalcon\Mvc\Model\Resultset::getFirst

### ISSUE THREAD

**Issue by sergeyklay** [phalcon/cphalcon#15027]
The current implementation of `Phalcon\Mvc\Model\Resultset::getFirst` implies that method must return a `Phalcon\Mvc\ModelInterface` instance or NULL:
https://github.com/phalcon/cphalcon/blob/00fb495e8e8c97f2733d8364648fafb228916c76/phalcon/Mvc/Model/Resultset.zep#L311-L320

But in some case, it MAY return a `Phalcon\Mvc\Model\Row` instance:
```php
$robots = new Robots();
$manager = $model->getModelsManager();

/** @var \Phalcon\Mvc\Model\Query $query */
$query = $manager->createQuery('SELECT r.id FROM Robots AS r');

/** @var \Phalcon\Mvc\Model\Resultset\Simple $result */
$result = $query->execute();

/** @var \Phalcon\Mvc\Model\Row $result */
$result->getFirst();
```

This leads to a fatal error:
```
PHP Fatal error:
Return value of Phalcon\Mvc\Model\Resultset::getFirst()
must implement interface Phalcon\Mvc\ModelInterface or be null,
instance of Phalcon\Mvc\Model\Row returned in Unknown on line 0
```

**Comment by niden** [comment]
Implemented in v4.0.x

**Comment by konkov-alexey** [comment]
the same problem with getLast() method in 4.0.6
